### PR TITLE
Submitting pseudocode

### PR DESCRIPTION
**Pull request by cody-berry**

### CODE REVIEW

File: 2748. Number of Beautiful Pairs/solution.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                
+                # get the last digit of the number.
+                
+                # if the digits have a gcd() of 1, increase the number of 
+                # beautiful pairs.
+        
+    # This is a replacement for the standard gcd(), or Greatest Common Divisor 
+    # function passed in two integers and returning the greatest divisor that 
+    # divides them both.
+    def gcd(self, a: int, b: int) -> int:
```
Comment (winry-fruitiwi) [inline 2748. Number of Beautiful Pairs/solution.py:27]: Hmm you are probably correct making a GCD function here. I was too lazy to do that.

File: 2748. Number of Beautiful Pairs/solution.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        # define the divisor that we're considering, starting from the smaller
+        # of them.
+        
+        # while the divisor is greater than 0...
+        
+            # if both a and b can be divided by those, return the divisor.
+            
+            # decrement the divisor.
+
+        # there is a garuntee we will find something because both a and b will
```
Comment (winry-fruitiwi) [inline 2748. Number of Beautiful Pairs/solution.py]: nit: guarantee, not garuntee. Would have expected garentee from you!

File: 2748. Number of Beautiful Pairs/solution.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+class Solution:
+    # Given a list of nums, it returns the number of beautiful pairs. A pair of
```
Comment (winry-fruitiwi) [inline 2748. Number of Beautiful Pairs/solution.py]: nit: what is it?

**Review by winry-fruitiwi — APPROVED** [review]
Just a few typos to fix, otherwise your algorithm is clever and not lazy at all.

File: 2748. Number of Beautiful Pairs/solution.py (hunk 3 above)
Comment (aeruru) [reply 2748. Number of Beautiful Pairs/solution.py]: you should just say, "this function" or just "return the number..."

File: 2748. Number of Beautiful Pairs/solution.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    # numbers is called "beautiful" if the greatest common divisor of the first
+    # digit of the first number and the last digit of the second number have a 
+    # greatest common divisor of 1, which is exactly what the second function 
+    # handles.
+    def countBeautifulPairs(self, nums: List[int]) -> int:
+        # define the number of beautiful pairs.
+        
+        # define an alternate nums list that's a tuple of the first and last
```
Comment (aeruru) [inline 2748. Number of Beautiful Pairs/solution.py]: What's an alternate nums list?

**Pull request merged**